src/toDoList.js

```diff
@@ -2,39 +2,153 @@ import React from 'react';
+    } else {
+      newItems = [...this.state.items, inputText]
+      searchItems = [...newItems]
+    }
+    //this.state.value = '';  dont change state variables directly... use setState instead
     this.setState({
       value : '',
-      items : this.state.items
+      items : newItems,
+      isUpdate : false,
+      id : 9999,
+      searchResults: searchItems
     });
   }
 
-  displayList(){
-    return (this.state.items.slice(0).reverse().map((item,i) => <li key={i}>{item}</li>));
+  removeItem(id) {
+    let isCheck = false;
+    let newItems = this.state.items.filter((item, i) => {
+      if(item === this.state.searchResults[id] && !isCheck) {
+        isCheck = true;
+      } else {
+        return item;
+      }
+    });
+    let searchItems = this.state.searchResults.filter((item, i) => i !== id);
+    this.setState({
+      items : newItems,
+      searchResults: searchItems
+    });
+  }
+
+  displayList() {
+    return (<ul>{this.state.searchResults.map((item, i) => {
+      return (
+        <li key={i}>{item}
+        <button
+          onClick={this.removeItem.bind(this, i)}
+          type="button"
+          className="btn btn-primary">Remove
+        </button>
+        <button
+          onClick={this.editItem.bind(this, i)}
+          type="button"
+          className="btn btn-primary">Edit
+        </button>
+      </li>);
+    })}</ul>);
+  }
+
+  doRenderAddOrUpdateButton() {
+    let buttonText = "Add";
+    if(this.state.isUpdate){buttonText = "Edit";}
+    return (
+      <button
+        onClick={this.addItem}
+        type="button"
+        className="btn btn-primary">{buttonText}
+      </button>
+    );
   }
 
-  render() {
-    let items = this.state.items;
+  doRenderSearchBar() {
+    return (
+      <input
+      className="input"
+      size="40"
+      type="text"
+      placeholder="Search..."
+      onChange={this.onChangeFunctions('search')}
+      value={this.state.searchValue}
+    />
+    );
+  }
+
+  onChangeFunctions(action) {
+    return (
+      action === 'search' ?
+      e => {
+        let searchResults = null;
+        searchResults = this.state.items.filter((value, i) => value.includes(e.target.value));
+        return this.setState(
+          {
+            searchValue: e.target.value,
+            searchResults: searchResults
+        })
+      } :
+      e => this.setState({value: e.target.value})
+    )
+  }
+
+  render() { //render is a react component method.
     return (
       <div>
         <div id="addTask">
-          <input className="input" size="40" type="text" placeholder="Enter Task" onChange={e => this.setState({ value: e.target.value })} value={this.state.value}/>
-          <button onClick={this.addItem} type="button" className="btn btn-primary">Add</button>
+          <input
+            className="input"
+            size="40"
+            type="text"
+            placeholder="Enter Task"
+            onChange={this.onChangeFunctions()}
```

Comment:
Why are the calls to `onChangeFunctions` ambiguous? If you choose to not provide arguments sometimes then provide default params in function definition.

Reply:
separated out functions because search bar is created as a different component.